core-java-modules/core-java/src/main/java/com/baeldung/iterationcounter/ForLoopIterationCounter.java

```diff
@@ -0,0 +1,49 @@
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ForLoopIterationCounter {
+    private final List<String> colors = Arrays.asList("red", "blue", "yellow", "green");
+
+    private void forEachCounter() {
+        int counter = 0;
+        for (String color : colors) {
+            counter++;
+            System.out.println("Value:" + color + ", iteration: " + counter);
+        }
+    }
+
+    public void listIterator() {
+        for (ListIterator<String> color = colors.listIterator(); color.hasNext(); ) {
```

Comment:
This is an ugly `for` loop as it has no last clause. It may be better expressed as a `while`. It's not a `forEach` loop

Reply:
should I then remove this section from the code and from the article?

Reply:
It could be provided as an example around a `while` loop if you think it adds something.